darasimiolaifa/amistad
# feat(refactor): refactor and organize project

## What does this PR do?
This PR implements code refactor and organization

## What major activities were carried out?
- create a handlers folder and separate the logic for handling the user and screams routes
- create a utils folder and a separate file for handling the authentication middleware
- create a validators file and put the input validation logic in it
- create a config file and put the firebase config object in it

## How can this be manually tested?
- Clone the repository
- Fetch the branch
- Install the dependencies
- Start project by running `firebase serve` in the terminal
- Copy the localhost url, append `/signup` to it, paste it in postman and send a `post` request to the endpoint using the schema: 
```json
{
	"email": "String (valid email)",
	"password": "String",
	"confirmPassword": "String",
	"handle": "String (must be unique)"
}
```
You should get a successful response.
```json
{
	"token": "String (valid jsonwebtoken)",
	"message": "`User ${userId} created successfully.`"
}
```
- Change the route from `/signup` to `/login` and send a post request with the schema containing the credentials you just registered with:
```json
{
	"email": "String",
        "password": "String"
}
```
You should get a successful response.
```json
{
	"token": "String (valid jsonwebtoken)"
}
```
- Copy the token response and set the `Authorization` header in postman to `Bearer Token`. Put the copied token into the token tab `without the quotes`
- Change the route from `/login` to `/screams` and send a post request with the schema:
```json
{
	"body": "String",
}
```
You should get a successful response.
```json
{
	"message": "`Document ${screamId} created successfully.`"
}
```
- Change the HTTP verb from `post` to `get` and send. You should get an array of screams as a response.
```json
[
    {
	"screamId": "String",
        "body": "String",
        "createdAt": "Datetime"
    },
]
```
